sacc/sacc.py

```diff
@@ -1200,12 +1195,13 @@ def add_ell_cl(self, data_type, tracer1, tracer2, ell, x,
         None
 
         """
-        if isinstance(window, BandpowerWindow):
-            if len(ell) != window.nv:
+        if isinstance(window, (BandpowerWindow, TopHatWindow)):
+            nv = window.nv if isinstance(window, BandpowerWindow) else len(window.min)
```

Comment:
I think this will crash if window.min is an integer?

Reply:
True, we must ensure `window.min` is an array. This will fail here but also [here](https://github.com/LSSTDESC/sacc/pull/83/files#diff-a8065f269e8fb2b9d2eb123f994563b490212041c8dc0f811302e3141221ecdcR181). 

The `min/max` class attributes must be arrays. We can use `np.asarray` within the `__init__` of `TopHatWindow` but I think this is related to the redesign/rethink of the window classes you mention.